### PR TITLE
feat(ui): add storage section to project resources card

### LINKED ISSUE #2240: Build LXD projects storage section (1-4 pools)
[Zeplin wireframe](https://app.zeplin.io/project/601454d5d1141e1b6ea271b7/screen/602fd58917b442a300a2a156)

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Built the initial version of the project storage section. Follow ups will include #2415 and #2432
- Added a selector to get a pod's storage pools sorted by default then id

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the project tab of a LXD KVM
- Check that the storage section of the card matches the [design](https://github.com/canonical-web-and-design/maas-ui/issues/2240) for 1-4 storage pools

## Fixes

Fixes #2240 

## Screenshot

Screenshot_2021-03-25 LXD project default bolla MAAS

### CODE REVIEW

File: ui/src/app/kvm/components/PodMeter/PodMeter.tsx [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 const PodMeter = ({
   allocated,
+  className,
   free,
+  inverted = false,
   segmented = false,
   unit = "",
 }: Props): JSX.Element | null => {
-  const total = allocated + free;
+  const total = Number((allocated + free).toPrecision(4));
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/components/PodMeter/PodMeter.tsx:22]: Driveby fix rounding weirdness. Apparently 5 + 2.81 = 7.8100000000000005

File: ui/src/app/store/pod/selectors.ts [hunk 2]
```diff
@@ -227,6 +240,34 @@ const getVmResource = createSelector(
...
+  [
+    (state: RootState, podId: Pod["id"]) =>
+      defaultSelectors.getById(state, podId),
+  ],
+  (pod) => {
+    if (!pod) {
+      return [];
+    }
+    const pools = pod.storage_pools || [];
+    return [...pools].sort((a, b) => {
```
Comment (huwshimi) [inline ui/src/app/store/pod/selectors.ts:259]: I don't think this spread is necessary here, Immer should guarantee that the store is not updated. Was this a precaution or was something odd going on?

**Review by huwshimi — APPROVED** [review]
Looks good. One comment.

File: ui/src/app/store/pod/selectors.ts (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/pod/selectors.ts:259]: This was just a copy+paste from within the original component so it might not be necessary any longer. Let me remove it and see if anything changes

Comment (Caleb-Ellis) [reply ui/src/app/store/pod/selectors.ts]: Seems to be fine!
